docs/excel/custom-functions-batching.md

```diff
@@ -0,0 +1,144 @@
+
+    Next, the `_makeRemoteRequest` function creates the `requestBatch` array, which contains just the raw information from the invocationEntries that needs to be processed by the API.
+
+    To give a visual example, the following diagram shows how the information from the `batch` array has been copied into the `batchCopy` array and the stock ticker string information separated out to the `requestBatch` array. The `batch` array has been emptied and could continue to fill with requests in the meantime while the `requestBatch` is processing.
+
+    ![The batch array is copied and emptied.](../images/Batch2.svg "Batching diagram")
+
+    To fill in the third `TODO` item, add the following code.
+
+    ```js
+    function _makeRemoteRequest() {
+        // Copy the shared batch and allow the building of a new batch while you are waiting for a response.
+
+        var batchCopy = _batch.slice();
+        _batch = [];
+        _isBatchScheduled = false;
+        // Build a simpler request batch that only contains the arguments for each invocation.
+
+        var requestBatch = [];
+        for (var i = 0; i < batchCopy.length; i++) {
+            requestBatch[i] = {
+                "operation": batchCopy[i].operation,
+                "arguments": batchCopy[i].arguments
+            };
+    }
+    ```
+
+* **TODO #4**
+
+    In the fourth `TODO`, you'll continue filling in the `_makeRemoteRequest` function.
+
+    The invocation entries from your `batchCopy` array still have unresolved Promises that need to be fulfilled. When the API returns the information that you can use to fulfill these Promises, you'll match up each response from the API to the Promises in the `batchCopy` array.
+
+    Fill in `TODO#4`with the following code:
+
+    ```js
+        // Make the remote request.
+
+        _fetchFromRemoteService(requestBatch)
+        .then(function (responseBatch) {
+        // Match each value from the response batch to its corresponding invocation entry from the request batch,
+        // and resolve the invocation Promise with its corresponding response value.
+
+            for (var i = 0; i < responseBatch.length; i++) {
+            batchCopy[i].resolve(responseBatch[i]);
+            }
+        });
+    }
+    ```
+
+    Through this process, the Promise for each invocation is fulfilled and each _pushOperation can be returned in the cell where the function was originally called. You can see in the diagram that follows that the API returned information to the `requestBatch` array, which resolved the Promises in the `batchCopy` array, which then provides the information for the return values of the original custom function invocations. The `batch` array is still empty, but could also begin filling with additional requests for the next batch to be processed.
```

Comment:
"in the following diagram"
"provides" should be "provided"

Reply:
I went with "Through this process, ...each is fulfilled. Each _pushOp returns" and then changed all the other tenses to present.

Reply:
Thanks!